File: apps/admin/src/api/item.ts
import { apiFetch } from './http'
import { buildListQuery } from './query'
import type {
  ItemCreateRep,
  ItemCreateReq,
  ItemListRep,
  ItemListReq,
  ItemMain,
  ItemUpdateReq,
} from './types'

/**
 * REST client for the kusec `Item` service (see api/proto/kusec_v1/item.proto).
 * Routes are singular per the gateway contract: `/item`, `/item/{id}`.
 * An Item is scoped to a Secret via `secret_id`.
 */

export function listItems(req: ItemListReq = {}): Promise<ItemListRep> {
  const query = buildListQuery(req.list_params, {
    secret_id: req.secret_id,
    secret_ids: req.secret_ids,
    active: req.active,
    search: req.search,
  })
  return apiFetch<ItemListRep>(`/item${query}`)
}

export function getItem(id: string): Promise<ItemMain> {
  return apiFetch<ItemMain>(`/item/${encodeURIComponent(id)}`)
}

export function createItem(req: ItemCreateReq): Promise<ItemCreateRep> {
  return apiFetch<ItemCreateRep>('/item', {
    method: 'POST',
    body: JSON.stringify(req),
  })
}

export function updateItem(id: string, req: ItemUpdateReq): Promise<void> {
  return apiFetch<void>(`/item/${encodeURIComponent(id)}`, {
    method: 'PUT',
    body: JSON.stringify(req),
  })
}

export function deleteItem(id: string): Promise<void> {
  return apiFetch<void>(`/item/${encodeURIComponent(id)}`, {
    method: 'DELETE',
  })
}

File: internal/mcpserver/client/model/item.go
package model

// Item — item секрета. Поле Value — чувствительное: за пределы клиента и
// value-механики MCP-сервера оно выходить не должно.
type Item struct {
	ID          string `json:"id"`
	CreatedAt   string `json:"created_at"`
	UpdatedAt   string `json:"updated_at"`
	SecretID    string `json:"secret_id"`
	Active      bool   `json:"active"`
	Key         string `json:"key"`
	Value       string `json:"value"`
	ValueFormat string `json:"value_format"`
	Encoding    string `json:"encoding"`
	FileName    string `json:"file_name"`
	ContentType string `json:"content_type"`
	Description string `json:"description"`
}

// List

type ItemListReq struct {
	ListParams ListParams
	SecretID   *string
	SecretIDs  []string
	Active     *bool
	Search     *string
}

type ItemListRep struct {
	PaginationInfo PaginationInfo `json:"pagination_info"`
	Results        []Item         `json:"results"`
}

// Create

type ItemCreateReq struct {
	SecretID    string `json:"secret_id"`
	Active      *bool  `json:"active,omitempty"`
	Key         string `json:"key"`
	Value       string `json:"value"`
	ValueFormat string `json:"value_format,omitempty"`
	Encoding    string `json:"encoding,omitempty"`
	FileName    string `json:"file_name,omitempty"`
	ContentType string `json:"content_type,omitempty"`
	Description string `json:"description"`
}

type ItemCreateRep struct {
	ID string `json:"id"`
}

// Update

type ItemUpdateReq struct {
	SecretID    *string `json:"secret_id,omitempty"`
	Active      *bool   `json:"active,omitempty"`
	Key         *string `json:"key,omitempty"`
	Value       *string `json:"value,omitempty"`
	ValueFormat *string `json:"value_format,omitempty"`
	Encoding    *string `json:"encoding,omitempty"`
	FileName    *string `json:"file_name,omitempty"`
	ContentType *string `json:"content_type,omitempty"`
	Description *string `json:"description,omitempty"`
}
